### ISSUE TITLE
hourly, monthly, & yearly workers

### ISSUE THREAD

**Issue by evilthreads669966** [evilthreads669966/BootLaces#100]
2020-12-31 20:35:20.074 4776-4853/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 20:35:20.085 4776-4850/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 20:35:20.125 4776-4845/com.candroid.lacedboots D/Hourly worker: working hourly

**Comment by evilthreads669966** [comment]
2020-12-31 23:36:35.260 19697-19758/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:36:35.265 19697-19743/com.candroid.lacedboots D/OneTimeWorker: working one time
2020-12-31 23:36:35.293 19697-19747/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:36:35.295 19697-19758/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:36:40.309 19697-19751/com.candroid.lacedboots D/Future worker: working
2020-12-31 23:36:40.313 19697-19751/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:37:04.878 19697-19749/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:37:04.880 19697-19744/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:37:14.891 19697-19753/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:37:14.893 19697-19752/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:37:24.902 19697-19747/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:37:24.905 19697-19744/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:37:34.920 19697-19758/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:37:34.922 19697-19750/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:37:44.951 19697-19752/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:38:44.971 19697-19747/com.candroid.lacedboots D/Hourly worker: working hourly

**Comment by evilthreads669966** [comment]
2020-12-31 23:36:35.260 19697-19758/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:36:35.265 19697-19743/com.candroid.lacedboots D/OneTimeWorker: working one time
2020-12-31 23:36:35.293 19697-19747/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:36:35.295 19697-19758/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:36:40.309 19697-19751/com.candroid.lacedboots D/Future worker: working
2020-12-31 23:36:40.313 19697-19751/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:37:04.878 19697-19749/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:37:04.880 19697-19744/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:37:14.891 19697-19753/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:37:14.893 19697-19752/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:37:24.902 19697-19747/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:37:24.905 19697-19744/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:37:34.920 19697-19758/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:37:34.922 19697-19750/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:37:44.951 19697-19752/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:38:44.971 19697-19747/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:39:45.035 19697-19748/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:39:45.037 19697-19759/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:39:55.080 19697-19996/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:40:55.126 19697-19996/com.candroid.lacedboots D/Daily worker: working daily
2020-12-31 23:40:55.126 19697-20093/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:41:05.142 19697-19751/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:41:55.188 19697-19751/com.candroid.lacedboots D/Daily worker: working daily
2020-12-31 23:41:55.191 19697-19748/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:42:15.187 19697-20093/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:42:25.237 19697-19759/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:42:35.259 19697-19759/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:42:35.260 19697-20093/com.candroid.lacedboots D/Monthly worker: working monthly
2020-12-31 23:42:45.276 19697-19751/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:42:45.278 19697-19752/com.candroid.lacedboots D/Monthly worker: working monthly
2020-12-31 23:42:55.324 19697-19751/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:43:05.383 19697-19996/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:43:05.390 19697-19758/com.candroid.lacedboots D/Monthly worker: working monthly
2020-12-31 23:43:15.429 19697-19759/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:43:15.430 19697-19996/com.candroid.lacedboots D/Monthly worker: working monthly
2020-12-31 23:43:25.483 19697-20092/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:44:25.486 19697-19751/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:44:35.505 19697-20092/com.candroid.lacedboots D/Periodic worker: working

**Comment by evilthreads669966** [comment]
2020-12-31 23:36:35.260 19697-19758/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:36:35.265 19697-19743/com.candroid.lacedboots D/OneTimeWorker: working one time
2020-12-31 23:36:35.293 19697-19747/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:36:35.295 19697-19758/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:36:40.309 19697-19751/com.candroid.lacedboots D/Future worker: working
2020-12-31 23:36:40.313 19697-19751/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:37:04.878 19697-19749/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:37:04.880 19697-19744/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:37:14.891 19697-19753/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:37:14.893 19697-19752/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:37:24.902 19697-19747/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:37:24.905 19697-19744/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:37:34.920 19697-19758/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:37:34.922 19697-19750/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:37:44.951 19697-19752/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:38:44.971 19697-19747/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:39:45.035 19697-19748/com.candroid.lacedboots D/Hourly worker: working hourly
2020-12-31 23:39:45.037 19697-19759/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:39:55.080 19697-19996/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:40:55.126 19697-19996/com.candroid.lacedboots D/Daily worker: working daily
2020-12-31 23:40:55.126 19697-20093/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:41:05.142 19697-19751/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:41:55.188 19697-19751/com.candroid.lacedboots D/Daily worker: working daily
2020-12-31 23:41:55.191 19697-19748/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:42:15.187 19697-20093/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:42:25.237 19697-19759/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:42:35.259 19697-19759/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:42:35.260 19697-20093/com.candroid.lacedboots D/Monthly worker: working monthly
2020-12-31 23:42:45.276 19697-19751/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:42:45.278 19697-19752/com.candroid.lacedboots D/Monthly worker: working monthly
2020-12-31 23:42:55.324 19697-19751/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:43:05.383 19697-19996/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:43:05.390 19697-19758/com.candroid.lacedboots D/Monthly worker: working monthly
2020-12-31 23:43:15.429 19697-19759/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:43:15.430 19697-19996/com.candroid.lacedboots D/Monthly worker: working monthly
2020-12-31 23:43:25.483 19697-20092/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:44:25.486 19697-19751/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:44:35.505 19697-20092/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:45:35.506 19697-20093/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:45:35.511 19697-20092/com.candroid.lacedboots D/Monthly worker: working monthly
2020-12-31 23:45:45.516 19697-19752/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:45:45.517 19697-19758/com.candroid.lacedboots D/Monthly worker: working monthly
2020-12-31 23:45:55.561 19697-19759/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:45:55.563 19697-19752/com.candroid.lacedboots D/Monthly worker: working monthly
2020-12-31 23:46:05.610 19697-19748/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:46:15.634 19697-19997/com.candroid.lacedboots D/Monthly worker: working monthly
2020-12-31 23:46:15.634 19697-19758/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:46:25.678 19697-20093/com.candroid.lacedboots D/Monthly worker: working monthly
2020-12-31 23:46:25.679 19697-19748/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:46:35.730 19697-19758/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:46:35.731 19697-19996/com.candroid.lacedboots D/Monthly worker: working monthly
2020-12-31 23:46:35.738 19697-19716/com.candroid.lacedboots W/roid.lacedboot: Suspending all threads took: 6.019ms
2020-12-31 23:46:45.753 19697-19996/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:46:55.768 19697-19758/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:46:55.771 19697-20093/com.candroid.lacedboots D/Monthly worker: working monthly
2020-12-31 23:47:05.815 19697-19997/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:47:05.819 19697-19759/com.candroid.lacedboots D/Monthly worker: working monthly
2020-12-31 23:47:15.848 19697-19997/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:47:25.883 19697-19996/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:47:25.891 19697-19751/com.candroid.lacedboots D/Monthly worker: working monthly
2020-12-31 23:47:35.931 19697-20092/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:48:35.898 19697-20093/com.candroid.lacedboots D/Monthly worker: working monthly
2020-12-31 23:48:35.899 19697-19759/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:48:45.932 19697-19752/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:48:45.942 19697-19759/com.candroid.lacedboots D/Monthly worker: working monthly
2020-12-31 23:49:45.913 19697-19752/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:49:45.915 19697-19759/com.candroid.lacedboots D/Monthly worker: working monthly
2020-12-31 23:49:55.927 19697-19752/com.candroid.lacedboots D/Periodic worker: working
2020-12-31 23:50:55.959 19697-20093/com.candroid.lacedboots D/Periodic worker: working

**Comment by evilthreads669966** [comment]
### this looks better

2021-01-01 17:11:46.034 7975-8001/com.candroid.lacedboots D/OneTimeWorker: working one time
2021-01-01 17:11:46.041 7975-8004/com.candroid.lacedboots D/Hourly Worker: working hourly
2021-01-01 17:11:46.047 7975-8004/com.candroid.lacedboots D/Hourly Worker: working hourly
2021-01-01 17:11:46.050 7975-8001/com.candroid.lacedboots D/Hourly Worker: working hourly
2021-01-01 17:11:46.115 7975-8006/com.candroid.lacedboots D/Future Worker: working
2021-01-01 17:11:46.123 7975-8006/com.candroid.lacedboots D/Hourly Worker: working hourly
2021-01-01 17:12:08.890 7975-8003/com.candroid.lacedboots D/Hourly Worker: working hourly
2021-01-01 17:12:08.893 7975-8006/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:12:08.894 7975-8006/com.candroid.lacedboots D/Daily Worker: working daily
2021-01-01 17:12:36.724 7975-8008/com.candroid.lacedboots D/Daily Worker: working daily
2021-01-01 17:12:36.726 7975-8008/com.candroid.lacedboots D/Monthly Worker: working monthly
2021-01-01 17:12:36.727 7975-8001/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:12:46.730 7975-8007/com.candroid.lacedboots D/Monthly Worker: working monthly
2021-01-01 17:12:46.733 7975-8008/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:12:46.734 7975-8003/com.candroid.lacedboots D/Yearly Worker: working yearly

**Comment by evilthreads669966** [comment]
persistent worker doesn't have a logging statement and needs to log now that it doesn't have a progress notif.
dumpsys on the alarms will get posted next

**Comment by evilthreads669966** [comment]
this looks to be working now and I guess to ensure that it is I need to somehow get the alarm that is set after the first occurrence of daily weekly monthly and yearly. which is done with adb dumpsys

**Comment by evilthreads669966** [comment]
the problem is that they are firing off twice ten seconds apart, that is the problem here but they should only happen once in the log file

**Comment by evilthreads669966** [comment]
## i think this is specifically an hourly worker problem as it is definitely not working correctly

```
2021-01-01 17:11:46.034 7975-8001/com.candroid.lacedboots D/OneTimeWorker: working one time
2021-01-01 17:11:46.041 7975-8004/com.candroid.lacedboots D/Hourly Worker: working hourly
2021-01-01 17:11:46.047 7975-8004/com.candroid.lacedboots D/Hourly Worker: working hourly
2021-01-01 17:11:46.050 7975-8001/com.candroid.lacedboots D/Hourly Worker: working hourly
2021-01-01 17:11:46.115 7975-8006/com.candroid.lacedboots D/Future Worker: working
2021-01-01 17:11:46.123 7975-8006/com.candroid.lacedboots D/Hourly Worker: working hourly
2021-01-01 17:12:08.890 7975-8003/com.candroid.lacedboots D/Hourly Worker: working hourly
2021-01-01 17:12:08.893 7975-8006/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:12:08.894 7975-8006/com.candroid.lacedboots D/Daily Worker: working daily
2021-01-01 17:12:36.724 7975-8008/com.candroid.lacedboots D/Daily Worker: working daily
2021-01-01 17:12:36.726 7975-8008/com.candroid.lacedboots D/Monthly Worker: working monthly
2021-01-01 17:12:36.727 7975-8001/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:12:46.730 7975-8007/com.candroid.lacedboots D/Monthly Worker: working monthly
2021-01-01 17:12:46.733 7975-8008/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:12:46.734 7975-8003/com.candroid.lacedboots D/Yearly Worker: working yearly
2021-01-01 17:12:56.741 7975-8008/com.candroid.lacedboots D/Yearly Worker: working yearly
2021-01-01 17:12:56.743 7975-8008/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:13:06.756 7975-8008/com.candroid.lacedboots D/Weekly Worker: working weekly
2021-01-01 17:13:06.771 7975-8003/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:13:16.786 7975-8003/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:14:16.771 7975-8003/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:15:16.771 7975-8003/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:16:17.720 7975-8003/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:17:56.796 7975-8007/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:18:16.785 7975-8009/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:19:16.785 7975-8003/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:20:57.812 7975-8007/com.candroid.lacedboots D/Hourly Worker: working hourly
2021-01-01 17:20:57.815 7975-8003/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:22:01.734 7975-8003/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:22:16.786 7975-8205/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:23:37.816 7975-8205/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:24:16.786 7975-8003/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:25:16.786 7975-8205/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:26:17.835 7975-8003/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:27:16.779 7975-8003/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:28:57.834 7975-8205/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:29:16.788 7975-8003/com.candroid.lacedboots D/Periodic Worker: working
2021-01-01 17:30:16.771 7975-8003/com.candroid.lacedboots D/Periodic Worker: working
```

**Comment by evilthreads669966** [comment]
this might be the service shutting down when no workers are working causing there to be multiple alarms scheduled for the same task

**Comment by evilthreads669966** [comment]
However there is supposed to be a persistent worker in the background meaning there would never be a situation where no worker is present in the current testing environment

**Comment by evilthreads669966** [comment]
adb shell dumpsys alarm
```
Batch{efc50de num=3 start=178200077 end=184568380}:
    RTC_WAKEUP #2: Alarm{bc642bf type 0 when 1609719708402 com.candroid.lacedboots}
      tag=*walarm*:com.candroid.lacedboots/com.candroid.bootlaces.BootReceiver
      type=0 expectedWhenElapsed=+2d0h7m40s200ms expectedMaxWhenElapsed=+25d6h7m40s200ms whenElapsed=+2d0h7m40s200ms maxWhenElapsed=+25d6h7m40s200ms when=2021-01-04 00:21:48.402
      window=+23d6h0m0s0ms repeatInterval=2678400000 count=0 flags=0x0
      operation=PendingIntent{a7acd8c: PendingIntentRecord{7325cd5 com.candroid.lacedboots broadcastIntent}}
    ELAPSED_WAKEUP #1: Alarm{a789cea type 2 when 107288369 com.google.android.gms}
      tag=*walarm*:com.google.android.gms.fido.authenticator.autoenroll.FIDO_KEY_VALIDITY_CHECK_DELAY_COMPLETE
      type=2 expectedWhenElapsed=+1d4h25m48s492ms expectedMaxWhenElapsed=+2d2h46m35s174ms whenElapsed=+1d4h25m48s492ms maxWhenElapsed=+2d2h46m35s174ms when=+1d4h25m48s492ms
      window=+22h20m46s682ms repeatInterval=0 count=0 flags=0x0
      operation=PendingIntent{c264bdb: PendingIntentRecord{dcd3678 com.google.android.gms/com.google.android.gms.fido startService}}
    ELAPSED_WAKEUP #0: Alarm{fc8f151 type 2 when 105478768 com.google.android.gms}
      tag=*walarm*:com.google.android.gms.auth.authzen.CHECK_REGISTRATION
      type=2 expectedWhenElapsed=+1d3h55m38s891ms expectedMaxWhenElapsed=+2d1h53m48s503ms whenElapsed=+1d3h55m38s891ms maxWhenElapsed=+2d1h53m48s503ms when=+1d3h55m38s891ms
      window=+21h58m9s612ms repeatInterval=0 count=0 flags=0x0
      operation=PendingIntent{7bb9db6: PendingIntentRecord{5b3cab7 com.google.android.gms/com.google.android.gms.auth_authzen__phone broadcastIntent}}
Batch{5db4224 num=1 start=437420095 end=891020095}:
    RTC_WAKEUP #0: Alarm{85a518d type 0 when 1609978928420 com.candroid.lacedboots}
      tag=*walarm*:com.candroid.lacedboots/com.candroid.bootlaces.BootReceiver
      type=0 expectedWhenElapsed=+5d0h8m0s218ms expectedMaxWhenElapsed=+10d6h8m0s218ms whenElapsed=+5d0h8m0s218ms maxWhenElapsed=+10d6h8m0s218ms when=2021-01-07 00:22:08.420
      window=+5d6h0m0s0ms repeatInterval=604800000 count=0 flags=0x0
      operation=PendingIntent{35bdf42: PendingIntentRecord{b305b53 com.candroid.lacedboots broadcastIntent}}
Batch{49c5c90 num=1 start=2544832074 end=2544832074 flgs=0x5}:
    RTC_WAKEUP #0: Alarm{2277989 type 0 when 1612086340399 com.google.android.gms}
      tag=*walarm*:com.google.android.gms.reminders.notification.ACTION_REFRESH_ALIAS_LOCATION_REMINDERS
      type=0 expectedWhenElapsed=+29d9h31m32s197ms expectedMaxWhenElapsed=+29d9h31m32s197ms whenElapsed=+29d9h31m32s197ms maxWhenElapsed=+29d9h31m32s197ms when=2021-01-31 09:45:40.399
      window=0 repeatInterval=0 count=0 flags=0x5
      operation=PendingIntent{accad8e: PendingIntentRecord{fdb59af com.google.android.gms/com.google.android.gms.reminders startService}}
Batch{212b1bc num=1 start=30331810084 end=53983810084}:
    RTC_WAKEUP #0: Alarm{7852545 type 0 when 1639873318409 com.candroid.lacedboots}
      tag=*walarm*:com.candroid.lacedboots/com.candroid.bootlaces.BootReceiver
      type=0 expectedWhenElapsed=+351d0h7m50s207ms expectedMaxWhenElapsed=+624d18h7m50s207ms whenElapsed=+351d0h7m50s207ms maxWhenElapsed=+624d18h7m50s207ms when=2021-12-19 00:21:58.409
      window=+273d18h0m0s0ms repeatInterval=31536000000 count=0 flags=0x0
      operation=PendingIntent{d6a149a: PendingIntentRecord{18de1cb com.candroid.lacedboots broadcastIntent}}



    +653ms running, 73 wakeups, 73 alarms: u0a159:com.candroid.lacedboots
      *walarm*:com.candroid.lacedboots/com.candroid.bootlaces.BootReceiver
```

**Comment by evilthreads669966** [comment]
closed in pull request #130
